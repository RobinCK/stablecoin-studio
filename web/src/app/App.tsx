/* eslint-disable @typescript-eslint/no-unused-vars */
import { I18nextProvider } from 'react-i18next';
import { ChakraProvider } from '@chakra-ui/react';
import { Provider } from 'react-redux';
import i18n from '../i18n';
import store from '../store/store';
import theme from '../theme/Theme';
import Router from '../Router/Router';
import { BrowserRouter } from 'react-router-dom';
import { Fonts } from '../components/Fonts';
import { Focus } from '../components/Focus';
import { ScrollBar } from '../components/Scrollbar';
import { useEffect, useState } from 'react';
import SDKService from '../services/SDKService';

function App() {
	const [SDKInit, setSDKInit] = useState<boolean>(false);
	const [SDKInitialize, setSDKInitialize] = useState<boolean>(true);

	useEffect(() => {
		// instanceSDK();
	}, []);

	useEffect(() => {
		if (SDKInitialize) {
			setSDKInit(true);
		}
	}, [SDKInitialize]);

	// const instanceSDK = async () => {
	// 	const instance = SDKService.getInstance();
	// 	// await instance.init({
	// 	// 	onInit: () => {
	// 	// 		console.log('Init');
	// 	// 	},
	// 	// });
	// 	// instance.onWalletExtensionFound(() => {
	// 	// 	console.log('Extension found');
	// 	// 	setSDKInitialize(true);
	// 	// });
	// };

	return (
		<I18nextProvider i18n={i18n}>
			<Provider store={store}>
				<ChakraProvider theme={theme}>
					<BrowserRouter>
						<Focus />
						<Fonts />
						<ScrollBar />
						{SDKInit ? <Router /> : <></>}
					</BrowserRouter>
				</ChakraProvider>
			</Provider>
		</I18nextProvider>
	);
}

export default App;
