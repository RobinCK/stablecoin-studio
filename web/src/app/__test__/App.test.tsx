import { render } from '@testing-library/react';
import App from '../App';
import * as reactDeviceDetect from 'react-device-detect';

describe(`<${App.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<App />, {});

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should render correctly when is mobile', () => {
		Object.defineProperty(reactDeviceDetect, 'isMobile', { get: () => true });

		const component = render(<App />, {});

		const text = component.getByTestId('isMobile');
		expect(text).toHaveTextContent('This app is for desktop use only');

		expect(component.asFragment()).toMatchSnapshot();
	});
});
