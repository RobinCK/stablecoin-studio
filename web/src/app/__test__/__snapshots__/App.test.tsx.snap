// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1amrwrf"
  >
    <button
      class="chakra-button css-aev8ym"
      data-testid="modal-term-conditions-button"
      type="button"
    >
      Terms & Conditions
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
